UpComingMovies/app/src/main/java/com/example/douglas/upcomingmovies/utilities/NetworkUtils.java

```diff
@@ -0,0 +1,136 @@
+package com.example.douglas.upcomingmovies.utilities;
+
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.net.Uri;
+import android.util.Log;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
+
+/**
+ * Created by Douglas on 21/10/2017.
+ */
+
+/**
+ * This class is a module for our application meaning that isolates network functions from the rest
+ * of the application.
+ */
+public class NetworkUtils {
+    final static String UP_COMMING_MOVIES_BASE_URL = "https://api.themoviedb.org/3/movie/upcoming";

```

Comment:
Good job creating a separated class to handle network requests. There are some tools that could have made this easier, such as Retrofit. May I ask why you preferred to implement these by your own?

Reply:
As I replied Before, I should have use Retrofit instead of make my own. It could saved me time for other purposes and avoid some errors.  Retrofit seems to be pretty straight forward and very easy to work with. I would definetely use in future releases thank you :)